### PR TITLE
Added the "Fill Opacity" feature to the available toolbar items

### PR DESCRIPTION
**Pull request by Jared234**
Until now, if you wanted to change the **fill opacity**, you had to go to _tools -> pen/textmarker-> fill-opacity_. But this PR adds the functionality to reach the **fill opacity** option more easily by providing a new toolbar item, as requested in #4123. 

I also created a new icon. The color scheme was fitted to the already existing **fill** icon. If you don't like the icon or its colors and have any suggestions on how to improve it, just let me now and I will fix it.

### CODE REVIEW

**Comment by bhennion** [comment]
Thanks for the PR! This sounds like a useful addition.
Could you maybe post a screenshot or two? (so that people can see the icons in context and tell if they look off to them, without having to pull and build the PR)

I assume (although I did not try your PR out) clicking on the icon makes the preexisting dialog appear (the dialog you get from the menu entry), right?
We could also imagine having a drop-down slider near the toolbar, instead of a pop-up window. That'd be more work though.

**Comment by Jared234** [comment]
Sure, no problem. Attached are a few screenshots in all of the different icon themes. 

Yes, you are right. At the moment the icon only opens the already existing dialog. However, I think this is the desired solution. I already thought about a drop-down slider, but I don't think that would be the best approach. If you want to select opacity, it's great to have a real-time visualization of your adjustment. IMO, if we just added a dropdown slider, many people might be forced to go to _Tools -> Pen/Text Marker-> Fill Opacity_ to get the opacity just right. 

However, I might be missing something. What are your thoughts on this? 

CustomizeToolbars
OpacitySettings
ToolbarColorDark
ToolbarColorLight
ToolbarLucideDark
ToolbarLucideLight

**Review by rolandlo — APPROVED** [review]
I like the icons and the functionality. The only thing I'm not sure about, whether we should also add `PEN_FILL_OPACITY` to some of the toolbars. Currently only users that learned to customize the toolbar profit of the new feature.

**Comment by bhennion** [comment]
> However, I might be missing something. What are your thoughts on this?

I guess the drop-down menu could contain both a slider and a small previous box, perhaps even, if there's enough room, an editable display of the value. The challenge is to keep the drop-down box relatively small.

Now that I see the icon in context, I'm not sure it can be understood intuitively. Wouldn't some form of gauge, filled with purple with a linear gradient of transparency, and cursor on it be more intuitive? Something like this 
(The shade of purple is off and there's obviously room for improvement):
Screenshot_20221001_123605

**Comment by Jared234** [comment]
@rolandlo I could definitely add this to some of the default toolbars. But I'm also not sure how useful it would be. The question is how many people need to change the opacity so often that it would make sense to have it as the default icon.

@bhennion I agree that a slider in general could be an interesting idea. For example, you could also set the pen size with a slider instead of opening the font/pen size dialog. Correct me if I'm wrong, but I don't think this feature exists for icons yet. It might be worth opening another issue for it, as I think there may still be a need for discussion. Do you agree?
As for the icon: the icon I created was influenced by the default opacity icon design. Here's a list of icons that come up when you search for "opacity":  
image
But if you want me to look for other designs, I can give it another try.
I believe that your icon might be misinterpreted as a gradient tool. For example, in Photoshop this is the gradient icon:
b6c22-gradient-tool-selection

**Comment by rolandlo** [comment]
In general I would say that putting too much effort into the GUI (creating a dropdown-box with some features for a single tool) is probably not worth it, since the GUI is something we will need to rethink more globally after we have moved to Gtk 4.
About the icons I think that both ideas (current PR icons and @bhennion's suggestion) are interesting. Not sure how good that gradient would work for the Lucide icons though.

**Comment by bhennion** [comment]
Well, TBH, I don't quite like the icon, and i don't feel it reflects what clicking on it does. I'm happy to be outvoted though :)

**Comment by Jared234** [comment]
Should I create a poll regarding the icon designs in the _Discussions_ section?
This is my first contribution to this project and I am not sure if this feature is actively used.

**Comment by GithubUser699** [comment]
@Jared234
Thank you for implementing this feature!
Also, I really like the icons you've created. I think it fits the general look the icons in xournalpp have.

**Comment by bhennion** [comment]
I guess this can be merged as is. No reason to block this for an icon looks disagreement.
Merging in 24 hours unless someone objects.

**Comment by GithubUser699** [comment]
> I guess this can be merged as is. No reason to block this for an icon looks disagreement. Merging in 24 hours unless someone objects.

Also, https://github.com/xournalpp/xournalpp/issues/4123 can be closed after merge.

**Pull request merged**
